### PR TITLE
Backport to 0.1: FuturesUnordered: Do not poll the same future twice per iteration

### PR DESCRIPTION
**Pull request by krallin**
Same as #2333. The same issue exists in 0.1, so backporting it there
helps for code that is still using Futures 0.1 in some places.

### CODE REVIEW

File: src/stream/futures_unordered.rs [hunk 1]
```diff
@@ -353,7 +352,7 @@ impl<T> Stream for FuturesUnordered<T>
                 // * The future was extracted above (taken ownership). That way
                 //   if it panics we're guaranteed that the future is
                 //   dropped on this thread and doesn't accidentally get
-                //   dropped on a different thread (bad).
+                //   dropped on YIELD_EVERYa different thread (bad).
```
Comment (taiki-e) [inline src/stream/futures_unordered.rs:355]: looks like typo.

**Review by taiki-e — COMMENTED** [review]
Thanks! This looks good to me aside from a nit.

**Comment by krallin** [comment]
Whoops, sorry for that

**Review by taiki-e — APPROVED** [review]
Thanks!

**Comment by taiki-e** [comment]
Published in 0.1.31.

**Comment by krallin** [comment]
> Published in 0.1.31.

Thanks! =)

**Pull request merged**